pkgs/by-name/ma/maltego/package.nix

```diff
@@ -0,0 +1,81 @@
+{ lib
+, stdenv
+, fetchzip
+, jre
+, giflib
+, gawk
+, makeBinaryWrapper
+, icoutils
+, copyDesktopItems
+, makeDesktopItem
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "maltego";
+  version = "4.6.0";
+
+  src = fetchzip {
+    url = "https://downloads.maltego.com/maltego-v4/linux/Maltego.v${finalAttrs.version}.linux.zip";
+    hash = "sha256-q+1RYToZtBxAIDSiUWf3i/3GBBDwh6NWteHiK4VM1HY=";
+  };
+
+  postPatch = ''
+      substituteInPlace bin/maltego \
+            --replace /usr/bin/awk ${lib.getExe gawk}
+  '';
+
+    desktopItems = [
+    (makeDesktopItem {
+      name = finalAttrs.pname;
+      desktopName = "Maltego";
+      exec = finalAttrs.meta.mainProgram;
+      icon = finalAttrs.pname;
```

Comment:
I don't think the icon file is following pname. Also this meta programming is usually unecessary and just complicates things.

Reply:
We can replace those `finalAttrs.xxx` to the actual strings.

Reply:
That's usually less complex and more correct with overwrites.

Reply:
I would leave the finalAttrs where appropriate, putting more static data into places like exec `exec = finalAttrs.meta.mainProgram;` just doesn't make sense because you might forget to update them if the binary name ever changes. And meta.mainProgram is excatly for the purpose of keeping the binary name

Reply:
If I would like to update this info in a state where the PR is already merged, do I need to create a new one? If so, what should be the name of the new PR?

Reply:
Yes if a pr is merged it's merged the only way to change something then is to do a new pr.
For any changes to a package the commit should always be named `$pname: what ever you are doing` and for package updates we have the "special" syntax of `$pname: $old_version -> $new_version`